src/Core/Block.cpp

```diff
@@ -139,15 +139,17 @@ void Block::insert(size_t position, ColumnWithTypeAndName elem)
     if (elem.name.empty())
         throw Exception("Column name in Block cannot be empty", ErrorCodes::AMBIGUOUS_COLUMN_NAME);
 
-    for (auto & name_pos : index_by_name)
-        if (name_pos.second >= position)
-            ++name_pos.second;
-
-    auto [it, inserted] = index_by_name.emplace(elem.name, position);
+    auto [it, inserted] = index_by_name.emplace(elem.name, (size_t)-1);
```

Comment:
This will give compiler warning about "C-style cast".

Reply:
That's correct! I will use std::npos to replace (size_t)-1. Thanks for review.

Reply:
yes，use std::npos is better!